rusk/src/bin/main.rs

```diff
@@ -80,14 +81,32 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
             .with_genesis_timestamp(config.chain.genesis_timestamp())
             .with_mempool(config.mempool.into())
             .with_state_dir(state_dir)
-            .with_generation_timeout(config.chain.generation_timeout())
-            .with_gas_per_deploy_byte(config.chain.gas_per_deploy_byte())
-            .with_min_deployment_gas_price(
-                config.chain.min_deployment_gas_price(),
-            )
-            .with_min_deploy_points(config.chain.min_deploy_points())
-            .with_min_gas_limit(config.chain.min_gas_limit())
-            .with_block_gas_limit(config.chain.block_gas_limit());
+            .with_min_gas_limit(config.chain.min_gas_limit());
+
+        #[allow(deprecated)]
+        {
+            if let Some(gas_byte) = config.chain.gas_per_deploy_byte() {
+                warn!("[chain].gas_per_deploy_byte is deprecated, use [vm].gas_per_deploy_byte");
+                node_builder = node_builder.with_gas_per_deploy_byte(gas_byte);
+            }
+            if let Some(price) = config.chain.min_deployment_gas_price() {
+                warn!("[chain].min_deployment_gas_price is deprecated, use [vm].min_deployment_gas_price");
+                node_builder =
+                    node_builder.with_min_deployment_gas_price(price);
+            }
+            if let Some(timeout) = config.chain.generation_timeout() {
+                warn!("[chain].generation_timeout is deprecated, use [vm].generation_timeout");
+                node_builder = node_builder.with_generation_timeout(timeout);
+            }
+            if let Some(min) = config.chain.min_deploy_points() {
+                warn!("[chain].min_deploy_points is deprecated, use [vm].min_deploy_points");
+                node_builder = node_builder.with_min_deploy_points(min);
+            }
+            if let Some(limit) = config.chain.block_gas_limit() {
+                warn!("[chain].block_gas_limit is deprecated, use [vm].block_gas_limit");
+                node_builder = node_builder.with_block_gas_limit(limit);
+            }
+        }
```

Comment:
Shouldn't the respective fields and functions also be deprecated in 'src/bin/config/chain.rs'?